e2e-tests/upgrade-sharded/run

```diff
@@ -9,223 +9,223 @@ test_dir=$(realpath $(dirname $0))
-                    | grep -vc 'NAME' | awk '{print $1}') -eq 1 ]]; do
-        sleep 5
-    done
-    sleep 10
-
-    desc 'check images and generation after operator upgrade'
-    wait_for_running "${cluster}-rs0" "${CLUSTER_SIZE}" "false"
-    wait_for_running "${cluster}-cfg" "${CLUSTER_SIZE}" "false"
-    wait_cluster_consistency "${cluster}" "${CLUSTER_SIZE}"
-    check_applied_images "operator"
-    compare_generation "1" "statefulset" "${cluster}-rs0"
-    compare_generation "1" "statefulset" "${cluster}-cfg"
-    compare_generation "1" "deployment" "${cluster}-mongos"
-    compare_generation "1" "psmdb" "${cluster}"
-
-    desc 'patch psmdb images and upgrade'
-    kubectl_bin patch psmdb "${cluster}" --type=merge --patch '{
+	kubectl_bin delete -f "${src_dir}/deploy/crd.yaml" || true
+	kubectl_bin delete -f "${src_dir}/deploy/rbac.yaml" || true
+	create_namespace "${namespace}"
+	deploy_operator_gh
+
+	curl -s "https://raw.githubusercontent.com/percona/percona-server-mongodb-operator/${GIT_TAG}/deploy/secrets.yaml" >"${tmp_dir}/secrets.yaml"
+	kubectl_bin apply -f "${conf_dir}/client.yml" \
+		-f "${tmp_dir}/secrets.yaml"
+
+	local cr_yaml="${tmp_dir}/cr_${GIT_TAG}.yaml"
+	prepare_cr_yaml "${cr_yaml}"
+	apply_cluster "${cr_yaml}"
+	wait_for_running "${cluster}-rs0" "${CLUSTER_SIZE}" "false"
+	wait_for_running "${cluster}-cfg" "${CLUSTER_SIZE}" "false"
+	wait_cluster_consistency "${cluster}" "${CLUSTER_SIZE}"
+	run_mongos 'db.createUser({user: "myApp", pwd: "myPass", roles: [{ db: "myApp", role: "readWrite" }]})' \
+		"userAdmin:userAdmin123456@${cluster}-mongos.${namespace}"
+
+	run_mongos 'use myApp\n db.test.insert({ x: 100500 })' "myApp:myPass@${cluster}-mongos.${namespace}"
+
+	compare_generation "1" "statefulset" "${cluster}-rs0"
+	compare_generation "1" "statefulset" "${cluster}-cfg"
+	compare_generation "1" "psmdb" "${cluster}"
+
+	desc 'upgrade operator'
+	kubectl_bin apply -f "${src_dir}/deploy/crd.yaml"
+	kubectl_bin apply -f "${src_dir}/deploy/rbac.yaml"
+	kubectl_bin patch deployment "${OPERATOR_NAME}" \
+		-p'{"spec":{"template":{"spec":{"containers":[{"name":"'"${OPERATOR_NAME}"'","image":"'"${TARGET_IMAGE}"'"}]}}}}'
+	kubectl_bin rollout status deployment/"${OPERATOR_NAME}"
+
+	desc 'wait for operator upgrade'
+	until [[ $(kubectl_bin get pods --selector=name="${OPERATOR_NAME}" \
+		-o custom-columns='NAME:.metadata.name,IMAGE:.spec.containers[0].image' \
+		| grep -vc 'NAME' | awk '{print $1}') -eq 1 ]]; do
+		sleep 5
+	done
+	sleep 10
+
+	desc 'check images and generation after operator upgrade'
+	wait_for_running "${cluster}-rs0" "${CLUSTER_SIZE}" "false"
+	wait_for_running "${cluster}-cfg" "${CLUSTER_SIZE}" "false"
+	wait_cluster_consistency "${cluster}" "${CLUSTER_SIZE}"
+	check_applied_images "operator"
+	compare_generation "1" "statefulset" "${cluster}-rs0"
+	compare_generation "1" "statefulset" "${cluster}-cfg"
+	compare_generation "2" "deployment" "${cluster}-mongos"
```

Comment:
generation should not be changed after that operator image upgrade

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/782#discussion_r745554933